arquillian/resolver/src/main/java/org/wildfly/swarm/arquillian/resolver/ShrinkwrapArtifactResolvingHelper.java

```diff
@@ -110,16 +118,48 @@ public ArtifactSpec resolve(ArtifactSpec spec) {
     }
 
     @Override
-    public Set<ArtifactSpec> resolveAll(final Set<ArtifactSpec> specs) {
+    public Set<ArtifactSpec> resolveAll(final Set<ArtifactSpec> specs, boolean trasitive) {
         if (specs.isEmpty()) {
-
             return specs;
         }
 
+        MavenResolutionStrategy transitivityStrategy = ( trasitive ? TransitiveStrategy.INSTANCE : NonTransitiveStrategy.INSTANCE );
+
         resetListeners();
         final MavenResolvedArtifact[] artifacts =
                 withResolver(r -> r.resolve(specs.stream().map(ArtifactSpec::mavenGav).collect(Collectors.toList()))
-                        .withTransitivity()
+                        .using(new CombinedStrategy(
+                                new MavenResolutionStrategy() {
+                                    @Override
+                                    public TransitiveExclusionPolicy getTransitiveExclusionPolicy() {
+                                        return null;
+                                    }
+
+                                    @Override
+                                    public MavenResolutionFilter[] getResolutionFilters() {
+                                        return new MavenResolutionFilter[]{
+                                                new MavenResolutionFilter() {
+                                                    @Override
+                                                    public boolean accepts(MavenDependency dependency, List<MavenDependency> dependenciesForResolution, List<MavenDependency> dependencyAncestors) {
+                                                        if ( dependency.getVersion().equals( "1.0.1B" ) ) {
```

Comment:
Okay, squashed in another commit that should remove the printlns.

/me double-checks.

On Wed, Aug 31, 2016 at 3:47 PM, Bob McWhirter <bmcwhirt@redhat.com> wrote:

> I’m here, about to push -f a fix to my PR.
>
>
> On Wed, Aug 31, 2016 at 3:25 PM, Ken Finnigan <notifications@github.com>
> wrote:
>
>> In arquillian/resolver/src/main/java/org/wildfly/swarm/arquilli
>> an/resolver/ShrinkwrapArtifactResolvingHelper.java
>> <https://github.com/wildfly-swarm/wildfly-swarm/pull/123#discussion_r77055729>
>> :
>>
>> >          resetListeners();
>> >          final MavenResolvedArtifact[] artifacts =
>> >                  withResolver(r -> r.resolve(specs.stream().map(ArtifactSpec::mavenGav).collect(Collectors.toList()))
>> > -                        .withTransitivity()
>> > +                        .using(new CombinedStrategy(
>> > +                                new MavenResolutionStrategy() {
>> > +                                    @Override
>> > +                                    public TransitiveExclusionPolicy getTransitiveExclusionPolicy() {
>> > +                                        return null;
>> > +                                    }
>> > +
>> > +                                    @Override
>> > +                                    public MavenResolutionFilter[] getResolutionFilters() {
>> > +                                        return new MavenResolutionFilter[]{
>> > +                                                new MavenResolutionFilter() {
>> > +                                                    @Override
>> > +                                                    public boolean accepts(MavenDependency dependency, List<MavenDependency> dependenciesForResolution, List<MavenDependency> dependencyAncestors) {
>> > +                                                        if ( dependency.getVersion().equals( "1.0.1B" ) ) {
>>
>> if you've "left the building" I can fix
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/wildfly-swarm/wildfly-swarm/pull/123/files/3c19dcf3389daaf0060cc7169d7cc361c47fbb8e#r77055729>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAA-T_JE7ou5sek3ylWgqVdzucX8IhLEks5qldUMgaJpZM4Jx3g7>
>> .
>>
>
>